Add Linux Support via Bash Scripts + Batch Fixes + Git Simplification + README Updates

KeyofBlueS opened the pull request:
Hi!

First off, thank you so much for this awesome project, Amplitude 2016 Deluxe is an amazing piece of work, and I really appreciate the time and effort you've put into making it so accessible and moddable.

Highlights of this PR:
- Added Linux support by introducing Bash script equivalents for the existing Batch scripts.
- Fixed a few issues in some of the existing Batch scripts.
- Replaced Python-based Git usage in scripts with direct Git commands, removing the need for Python (since Git is already required for cloning the repo).
- Still updated the Python script with a few enhancements, just in case it's decided to keep using it.
- Updated the README to reflect cross-platform compatibility and the reduced dependency on Python.

Notes:
- I've only tested the Linux scripts for the PS3 version of the game. PS4 is untested, as I don't have the hardware.
- The fixes to the existing Batch scripts haven't been tested either.
- For now, running the Windows executables (AmpHelper.exe and DTXTool.exe) is possible via Wine. If I get the time, I might try building native Linux versions where possible.
- The "dev_scripts" doesn't yet have Linux versions. I'm unsure if I'll port those, but I'm open to doing it if there's demand.
- The Linux equivalent of the init script would be `_init_repo.sh`:
```
#!/bin/bash

cd "$(dirname "$0")"

git init
git pull 'https://github.com/hmxmilohax/amplitude-2016-deluxe' main

exit 0
```

I'd love it if you could take a look and let me know if everything checks out, or if you'd like any adjustments. Thanks again for all you do!

Best,
KeyofBlueS

KeyofBlueS committed:
- Create _build\ps4\AFR\CUSA02480 output directory.

- Removed trailing period (likely a typo) in amphelper song add-all command.

KeyofBlueS committed:
- Create _build\ps4\AFR\CUSA02480 output directory.

KeyofBlueS committed:
- Fixed the output directory path to ensure files are copied in the correct location.

KeyofBlueS committed:
- Use direct Git calls in this batch script:

Replaced Python-based Git execution with direct Git commands to eliminate the Python dependency, as Git is assumed to be already installed for cloning this repo.

KeyofBlueS committed:
- fix paths, add error handling, cleanup copy logic:

- Replaced hardcoded backslashes with Path joins for cross-platform support
- Added missing sys import for sys.exit
- Improved Git error handling using git.exc.GitCommandError
- Added check to remove existing destination folder before copy

KeyofBlueS committed:
- Added Bash script equivalents for the existing Batch script.

KeyofBlueS committed:
- Added Linux support instructions alongside existing Windows steps.

DarkRTA reviewed (approved):
lgtm but i want someone else with linux experience to look over these scripts before we merge this

i may have missed something while looking over it

jnackmclain commented:
i can test ps4 and linux tonight, ping me if i forget

kernaltrap8 commented on _build_ps3.sh:
```diff
@@ -0,0 +1,65 @@
+fi
+
+echo "Creating build directories..."
+mkdir -p "./_build/ps3/USRDIR"
```
Instead of using `.`, you could use `$PWD` for full paths. If you just want relative, you shouldn't need `.`. Everything else looks fine.

KeyofBlueS replied:
Thanks for the suggestion!
Using `.` instead of `$PWD` is intentional here, the script runs `cd "$(dirname "$0")"` at the start, so all relative paths are scoped to the script's directory and work reliably.

While `./` isn't strictly necessary for relative paths, I use it for clarity to indicate the path starts from the current directory, it's often considered good practice in scripts.

That said, if you'd prefer absolute paths using `$PWD`, I can absolutely make that change, just let me know if you'd like me to make this change explicitly. Obviously, this should be done in all the bash scripts, not just in `_build_ps3.sh`.

DarkRTA replied:
relative paths here are fine given the context 
i personally dont think this needs to be adjusted

kernaltrap8 commented on _build_ps3.sh:
```diff
@@ -0,0 +1,65 @@
+mkdir -p "./_build/ps3/USRDIR"
+mkdir -p "./_tmpbuild"
+
+git pull 'https://github.com/hmxmilohax/amplitude-2016-deluxe' main
```
I don't think this git command is correct.
```diff
- git pull 'https://github.com/hmxmilohax/amplitude-2016-deluxe' main
+ git clone 'https://github.com/hmxmilohax/amplitude-2016-deluxe' -b main
```

KeyofBlueS replied:
Actually, the original command was correct. `git pull` updates the current local repository by fetching and merging changes.
Also, if you try to run `git clone` inside an existing local repository, it will result in an error.

kernaltrap8 commented on _build_ps3.sh:
```diff
@@ -0,0 +1,65 @@
+done
+
+echo "Converting .script files..."
+find _tmpbuild -type f -iname "*.script" ! -iname "*.script_dta_ps3" | while read -r file; do
```
Directory passed to `find` should be changed to match line 34

KeyofBlueS replied:
Fair enough! Also in `_build_ps3_nosong.sh`, `_build_ps4.sh` and `_build_ps4_nosong.sh`.
Thanks!

kernaltrap8 commented on _build_ps3.sh:
```diff
@@ -0,0 +1,65 @@
+find "./_tmpbuild" -type f -iname "*.script" ! -iname "*.script_dta_ps3" -exec rm {} \;
+
+echo "Copying processed files to prep directory..."
+mkdir -p "._prep_ps3/ext_ark/ps3"
```
Typo?
```diff
- mkdir -p "._prep_ps3/ext_ark/ps3"
+ mkdir -p "./_prep_ps3/ext_ark/ps3"
```

KeyofBlueS replied:
Yes a typo! Also present in `_build_ps3_nosong.sh`, `_build_ps4.sh` and `_build_ps4_nosong.sh`.
Thanks again!

KeyofBlueS committed:
- Updated the directory argument in the find command for consistency.

- Corrected a typo.

KeyofBlueS committed:
- Updated the directory argument in the find command for consistency.

- Corrected a typo.

KeyofBlueS committed:
- Updated the directory argument in the find command for consistency.

- Corrected a typo.

KeyofBlueS committed:
- Updated the directory argument in the find command for consistency.

- Corrected a typo.

KeyofBlueS committed:
- Aligned the source directory argument in the cp command with other scripts for consistency.

DarkRTA commented:
thanks for fixing the typos. this should be good to go after jnack and/or lysix looks at this

KeyofBlueS committed:
- Moved @echo off to the top of the script to prevent REM lines from being displayed.

KeyofBlueS committed:
Fix xcopy can't tell if destination path is a file or a directory if the destination folder doesn't exist yet.

KeyofBlueS committed:
Fix xcopy can't tell if destination path is a file or a directory if the destination folder doesn't exist yet.

KeyofBlueS committed:
Fix xcopy can't tell if destination path is a file or a directory if the destination folder doesn't exist yet.

KeyofBlueS committed:
Fix xcopy can't tell if destination path is a file or a directory if the destination folder doesn't exist yet.

KeyofBlueS committed:
- Fix missing closing quote at the end of the destination path.

- Fix xcopy can't tell if destination path is a file or a directory if the destination folder doesn't exist yet.

KeyofBlueS committed:
Aligned the destination directory argument in the xcopy command with other scripts for consistency.

KeyofBlueS commented:
I had some time to test in a Windows environment, so I’ve added additional fixes to the existing batch scripts in the above commits.

KeyofBlueS commented:
Hey @jnackmclain! Just a lil' ping, you mentioned you'd test PS4 and Linux. Let me know how it goes when you get a chance!

LlysiX commented on _download_custom_songs.bat:
unsure why this was changed, the python script was made to be platform agnostic

LlysiX replied:
this is fine as a stopgap, but to add on to this, i think it may be beneficial to refactor the entire build process to be more in line with [rb3dx](https://github.com/hmxmilohax/rock-band-3-deluxe)/[gh2dxu](https://github.com/hmxmilohax/Guitar-Hero-II-Deluxe-Unified)

KeyofBlueS replied:
The main reason for the change was to remove the python dependency. Since the python script was just calling git, it felt cleaner and more lightweight to use git directly, especially given that git is already required anyway.

KeyofBlueS commented:
Hi @jnackmclain, just a friendly reminder that this PR is ready for your review whenever you have a moment. It's the last one needed before we can get it merged.

Thanks a lot!

KeyofBlueS commented:
Thanks for merging this, really appreciated!

Just a quick note: as mentioned in the PR description, the `_init_repo.sh` file should also be included in the release.
